homeassistant/components/wittiot/sensor.py

```diff
@@ -0,0 +1,558 @@
+        key="LEAK",
+        icon="mdi:water-alert",
+    ),
+    WittiotDataTypes.BATTERY: WittiotSensorEntityDescription(
+        key="BATTERY",
+        icon="mdi:battery",
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Wittiot sensor entities based on a config entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    sensors: list[MainDevWittiotSensor | SubDevWittiotSensor] = []
+    # Main Device Data
+    for desc in SENSOR_DESCRIPTIONS:
+        if coordinator.data.get(desc.key) not in ("", "--", "--.-", "None"):
+            sensors.append(
+                MainDevWittiotSensor(
+                    coordinator,
+                    entry.data[CONF_HOST],
+                    entry.data[DEVICE_NAME],
+                    desc,
+                )
+            )
+
+    # Subdevice Data
+    for key in coordinator.data:
+        if coordinator.data.get(key) not in ("", "--", "--.-", "None"):
+            if key in MultiSensorInfo.SENSOR_INFO:
+                mapping = WITTIOT_SENSORS_MAPPING[
+                    MultiSensorInfo.SENSOR_INFO[key]["data_type"]
+                ]
+                description = dataclasses.replace(
+                    mapping,
+                    key=key,
+                    sensor_type=MultiSensorInfo.SENSOR_INFO[key]["dev_type"],
+                    name=MultiSensorInfo.SENSOR_INFO[key]["name"],
+                )
+                sensors.append(
+                    SubDevWittiotSensor(
+                        coordinator,
+                        entry.data[CONF_HOST],
+                        entry.data[DEVICE_NAME],
+                        description,
+                    )
+                )
+
+    async_add_entities(sensors, True)
+
+
+class MainDevWittiotSensor(
+    CoordinatorEntity[WittiotDataUpdateCoordinator], SensorEntity
+):
+    """Define an Local sensor."""
+
+    _attr_has_entity_name = True
+    entity_description: WittiotSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: WittiotDataUpdateCoordinator,
+        host: str,
+        devname: str,
+        description: WittiotSensorEntityDescription,
+    ) -> None:
+        """Initialize."""
+        super().__init__(coordinator)
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, f"{devname}_{MAIN_DATA}")},
+            manufacturer="WittIOT",
+            name=f"{devname}_{MAIN_DATA}",
+            model=self.coordinator.data["ver"],
+            configuration_url=f"http://{host}",
+        )
+        self._attr_unique_id = f"{devname}_{description.key}"
+        self.entity_description = description
+
+    @property
+    def native_value(self) -> str | int | float | None:
+        """Return the state."""
+        state = self.coordinator.data[self.entity_description.key]
+        if state not in ("", "--", "--.-", "None"):
```

Comment:
Can you move these checks to the library?

Reply:
Okay, I'll move it to the library

Reply:
I have deleted the display of channel data in this PR.